homeassistant/components/efesto/climate.py

```diff
@@ -0,0 +1,290 @@
+    """Representation of a Efesto heating device."""
+
+    def __init__(self, client, name, device_id):
+        """Initialize the thermostat."""
+        self.client = client
+        self._device_id = device_id
+        self._on = False
+        self._device_status = None
+        self._idle_info = None
+        self._human_device_status = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._smoke_temperature = None
+        self._real_power = None
+        self._current_power = None
+        self._name = name if name else device_id
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        return {
+            ATTR_DEVICE_STATUS: self._device_status,
+            ATTR_HUMAN_DEVICE_STATUS: self._human_device_status,
+            ATTR_SMOKE_TEMP: self._smoke_temperature,
+            ATTR_REAL_POWER: self._real_power,
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._device_id
+
+    @property
+    def name(self):
+        """Return the name of the Efesto, if any."""
+        return self._name
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Micronova",
+            "model": "Efesto",
+        }
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode.
+
+        Need to be one of HVAC_MODE_*.
+        """
+        if self._on:
+            return HVAC_MODE_HEAT
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        return [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+
+    @property
+    def fan_mode(self):
+        """Return fan mode."""
+        if self._current_power in FAN_MODES:
+            return self._current_power
+        return FAN_1
```

Comment:
Why are you returning the integer `1` if `fan_mode` expects a string?
See https://developers.home-assistant.io/docs/en/entity_climate.html#properties

You will need to translate between the fan modes of efesto and HA, e.g.
https://github.com/home-assistant/home-assistant/blob/6f8f23238ad2db3acfc949cb497dca33dd604ab4/homeassistant/components/fibaro/climate.py#L34-L47